src/m365/spo/commands/group/group-member-remove.ts

```diff
@@ -80,19 +95,48 @@ class SpoGroupMemberRemoveCommand extends SpoCommand {
   }
 
   #initOptionSets(): void {
-    this.optionSets.push({ options: ['groupName', 'groupId'] });
+    this.optionSets.push(
+      { options: ['groupName', 'groupId'] },
+      { options: ['userName', 'email', 'userId'] }
+    );
+  }
+
+  private async getUserName(args: CommandArgs): Promise<string> {
+    if (args.options.userName) {
+      return args.options.userName;
+    }
+    else {
+      const options: AadUserGetCommandOptions = {
+        email: args.options.email,
+        output: 'json',
+        debug: args.options.debug,
+        verbose: args.options.verbose
+      };
+
+      const userGetOutput: CommandOutput = await Cli.executeCommandWithOutput(AadUserGetCommand as Command, { options: { ...options, _: [] } });
+      const userOutput = JSON.parse(userGetOutput.stdout);
+      return userOutput.userPrincipalName;
+    }
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
     const removeUserfromSPGroup: () => Promise<void> = async (): Promise<void> => {
       if (this.verbose) {
-        logger.logToStderr(`Removing User with Username ${args.options.userName} from Group: ${args.options.groupId ? args.options.groupId : args.options.groupName}`);
+        logger.logToStderr(`Removing User ${args.options.userName || args.options.email || args.options.userId} from Group: ${args.options.groupId ? args.options.groupId : args.options.groupName}`);
       }
 
-      const loginName: string = `i:0#.f|membership|${args.options.userName}`;
-      const requestUrl: string = `${args.options.webUrl}/_api/web/sitegroups/${args.options.groupId
+      let requestUrl: string = `${args.options.webUrl}/_api/web/sitegroups/${args.options.groupId
         ? `GetById('${args.options.groupId}')`
-        : `GetByName('${formatting.encodeQueryParameter(args.options.groupName as string)}')`}/users/removeByLoginName(@LoginName)?@LoginName='${formatting.encodeQueryParameter(loginName)}'`;
+        : `GetByName('${formatting.encodeQueryParameter(args.options.groupName as string)}')`}`;
+
+      if (args.options.userId) {
+        requestUrl += `/users/removeById(${args.options.userId})`;
+      }
+      else {
+        const userName: string = await this.getUserName(args);
+        const loginName: string = `i:0#.f|membership|${userName}`;
+        requestUrl += `/users/removeByLoginName(@LoginName)?@LoginName='${formatting.encodeQueryParameter(loginName)}'`;
+      }
```

Comment:
Could you fix the confirmation message by the way? It shows `undefined` when removing members by userId or email.

`message: `Are you sure you want to remove user User ${args.options.userName} from SharePoint group?``